File: nodejs/generatePassword.js
function generatePassword(length, includeSymbols = false) {
    // Define character set
    let characters = includeSymbols
      ? 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+~`|}{[]\\:;?><,./-='
      : 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
    // Generate password
    let password = '';
    for (let i = 0; i < length; i++) {
      password += characters.charAt(Math.floor(Math.random() * characters.length));
    }
    return password;
  }

// Generate a 12-character password without symbols
let password1 = generatePassword(12);
console.log(password1);

// Generate a 16-character password with symbols
let password2 = generatePassword(16, true);
console.log(password2);

File: python/generatePassword.py
import string
import random

def generate_password(length, include_symbols=False):
    """Generate a strong password of given length"""
    # Define character set
    if include_symbols:
        characters = string.ascii_letters + string.digits + string.punctuation
    else:
        characters = string.ascii_letters + string.digits
    # Generate password
    password = ''.join(random.choice(characters) for i in range(length))
    return password

# Generate a 12-character password without symbols
password1 = generate_password(12)
print(password1)

# Generate a 16-character password with symbols
password2 = generate_password(16, include_symbols=True)
print(password2)    